newton/_src/viewer/gl/opengl.py

```
@@ -1334,6 +1355,108 @@ def render(self, camera, objects, lines=None, wireframe_shapes=None, arrows=None
+        )
+        grid_w = cols * cell_w + max(0, cols - 1) * spacing_px
+        grid_h = rows * cell_h + max(0, rows - 1) * spacing_px
+        origin_x = (screen_w - grid_w) * 0.5
+        origin_y = (screen_h - grid_h) * 0.5
+
+        gl.glActiveTexture(gl.GL_TEXTURE0)
+        gl.glBindTexture(gl.GL_TEXTURE_2D, int(texture_id))
+        with self._frame_shader:
+            for i in range(tile_count):
+                display_row, display_col = divmod(i, cols)
+                draw_x = max(0, int(round(origin_x + display_col * (cell_w + spacing_px))))
```

Ruff (0.16.1)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)